### PR TITLE
Add reverse sns and twitter lookup

### PR DESCRIPTION
**Pull request by 0xavro**
Adding reverse sns and twitter search

reverse-sns-twitter-search

### CODE REVIEW

File: packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx [hunk 1]
```diff
@@ -307,6 +320,28 @@ export default function CreateThread({
...
+          if (program?.provider.connection) {
+            const pubKey = new anchor.web3.PublicKey(address);
+
+            const fetchPromises = [
+              tryGetName(program?.provider.connection, pubKey),
+              fetchSolanaNameServiceName(program?.provider.connection, address),
+            ];
+
+            // TODO: Fix typing on promise result
+            const result: any = await Promise.all(fetchPromises);
```
Comment (fsher) [inline packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx:334]: For better readability, let's destruct the array and name the results.
```
const [name, snsName] = await Promise.all([
  ... // no need for an extra variable
];

if (name && snsName) { ... }
```

File: packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx [hunk 2]
```diff
@@ -359,7 +394,14 @@ export default function CreateThread({
           onChange={(e) => onAddressChange(e.target.value)}
         />
         {!isTyping
-          ? showAddressResult(validAddress, address, cardinalAddress, snsAddress)
+          ? showAddressResult(
+              validAddress,
```
Comment (fsher) [inline packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx:398]: That's a lot of arguments :D

Might want to rethink the signature of that function later. No need to do anything now.

**Review by fsher — APPROVED** [review]
LGTM, left a minor comment, please check it out.

File: packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx (hunk 2 above)
Comment (0xavro) [reply packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx:398]: Yes, let's refactor this function I think some of the logic can be simplified in there.

File: packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx (hunk 1 above)
Comment (0xavro) [reply packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx:334]: Sounds good. let's go with `twitterName` and `snsName`.

**Pull request merged**